### PR TITLE
fix:(Core\Instance):Gothik souls travel to dead zone

### PR DESCRIPTION
**Pull request by Ragelezz**
## Changes Proposed:
-  
-  

## Issues Addressed:

- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4273

## Tests Performed:

- .go c id 16060 and engage Gothik, killed minions and their souls won't "travel" to dead zone
- 

## How to Test the Changes:

1. .go c id 16060 and engage Gothik, kill his minions and their souls now "travel" to dead zone
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp [hunk 1]
```diff
@@ -374,13 +374,16 @@ class boss_gothik : public CreatureScript
             switch (spellInfo->Id)
             {
                 case SPELL_INFORM_LIVING_TRAINEE:
-                    me->SummonCreature(NPC_DEAD_TRAINEE, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation());
+                    if(Creature* target = me->SummonCreature(NPC_DEAD_TRAINEE, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
+                        me->CastSpell(target, SPELL_INFORM_DEAD_TRAINEE, true);
```
Comment (Helias) [inline src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:378]:
```suggestion
                    if(Creature* target = me->SummonCreature(NPC_DEAD_TRAINEE, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
                    {
                        me->CastSpell(target, SPELL_INFORM_DEAD_TRAINEE, true);
                    }
```

add the brackets even for the other code https://www.azerothcore.org/wiki/cpp-code-standards#brackets

**Comment by Si1ker** [comment]
So, since I saw the changes proposed I knew this wouldn't address the issue, so I personally took the time to write this so you can see why I was not able to fix it by myself, is not an easy thing but you would know that if you have read the issue entirely. It requires the structure I linked from TrinityCore and new invisible triggers/stalkers into the pile of bones, again.. all of these are on the issue reported.

-----

1- Please read the issue, the complete issue report.

2- The soul is taken and redirected to the summoned minion.
- Wrong/This PR: A "living minion" dies, gothik get's the soul, at the same time the soul is redirected by gothik to the minion itself.
- Blizzlike: The soul should be taken, travel to the other end of the platform then redirected to the pile of bones, **only once it lands**, the minion should appear.

====

Here, it is just what I said before, the soul goes directly to the summon minion, the summoned minion is already active while the soul is traveling.
image

--

Same thing as before but with a different angle.
image
image


--

I have confirmed this is the behavior in retail:Shadownalds , it has been the same since Naxxramas came out, no changes have been done to that "summoning style" since woltk to Shadowlands.

I did not record my session during my testing at retail at that time but here are some clips so you can check what I'm talking about:

https://youtu.be/AOn8fwtKFqE?t=38
https://youtu.be/80JUgC2bB50?t=38
https://youtu.be/Y7f88Rf17Gc?t=42 -> check top right corner to see the traveling pattern on the platform

**Comment by Ragelezz** [comment]
Got your point, will do today but via C++ script not within db.

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp [hunk 2]
```diff
@@ -374,13 +374,18 @@ class boss_gothik : public CreatureScript
...
-                    me->SummonCreature(NPC_DEAD_TRAINEE, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation());
+                    if(Creature* target = me->SummonCreature(NPC_DEAD_TRAINEE, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
+                    {
+                        me->CastSpell(target, SPELL_INFORM_DEAD_TRAINEE, true);
+                    }
                     break;
                 case SPELL_INFORM_LIVING_KNIGHT:
-                    me->SummonCreature(NPC_DEAD_KNIGHT, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation());
+                    if (Creature* target = me->SummonCreature(NPC_DEAD_KNIGHT, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
+                        me->CastSpell(target, SPELL_INFORM_DEAD_KNIGHT, true);
```
Comment (Helias) [inline src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:384]: add the bracket even here and below

Comment (Helias) [reply src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:388]: fix indentation

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp [hunk 3]
```diff
@@ -374,15 +374,30 @@ class boss_gothik : public CreatureScript
...
+                        }
                     break;
+                    }
                 case SPELL_INFORM_LIVING_RIDER:
-                    me->SummonCreature(NPC_DEAD_RIDER, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation());
+                    {
+                    if (Creature* target = me->SummonCreature(NPC_DEAD_RIDER, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
+                        {
+                        me->CastSpell(target, SPELL_INFORM_DEAD_RIDER, true);
+                        }
```
Comment (Helias) [inline src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:397]:
```suggestion
                    if (Creature* target = me->SummonCreature(NPC_DEAD_RIDER, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
                    {
                        me->CastSpell(target, SPELL_INFORM_DEAD_RIDER, true);
                    }
```

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp [hunk 4]
```diff
@@ -374,15 +374,30 @@ class boss_gothik : public CreatureScript
...
                     me->SummonCreature(NPC_DEAD_HORSE, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation());
                     break;
+                    }
```
Comment (Helias) [inline src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:400]:
```suggestion
                }
```

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp (hunk 3 above)
Comment (Helias) [inline src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:397]:
```suggestion
                    if (Creature* target = me->SummonCreature(NPC_DEAD_RIDER, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
                    {
                        me->CastSpell(target, SPELL_INFORM_DEAD_RIDER, true);
                    }
```

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp [hunk 5]
```diff
@@ -374,15 +374,30 @@ class boss_gothik : public CreatureScript
...
+                        {
+                        me->CastSpell(target, SPELL_INFORM_DEAD_TRAINEE, true);
+                        }
                     break;
+                    }
                 case SPELL_INFORM_LIVING_KNIGHT:
-                    me->SummonCreature(NPC_DEAD_KNIGHT, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation());
+                    {
+                    if (Creature* target = me->SummonCreature(NPC_DEAD_KNIGHT, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
+                        {
```
Comment (Helias) [inline src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:387]:
```suggestion
                    {
```

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp [hunk 6]
```diff
@@ -374,15 +374,30 @@ class boss_gothik : public CreatureScript
...
+                        me->CastSpell(target, SPELL_INFORM_DEAD_KNIGHT, true);
+                        }
```
Comment (Helias) [inline src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:389]:
```suggestion
                    }
```

File: src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp [hunk 7]
```diff
@@ -374,15 +374,30 @@ class boss_gothik : public CreatureScript
             switch (spellInfo->Id)
             {
                 case SPELL_INFORM_LIVING_TRAINEE:
-                    me->SummonCreature(NPC_DEAD_TRAINEE, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation());
+                    {
+                    if (Creature* target = me->SummonCreature(NPC_DEAD_TRAINEE, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
+                        {
+                        me->CastSpell(target, SPELL_INFORM_DEAD_TRAINEE, true);
+                        }
```
Comment (Helias) [inline src/server/scripts/Northrend/Naxxramas/boss_gothik.cpp:381]:
```suggestion
                    if (Creature* target = me->SummonCreature(NPC_DEAD_TRAINEE, PosSummonDead[pos].GetPositionX(), PosSummonDead[pos].GetPositionY(), PosSummonDead[pos].GetPositionZ(), PosSummonDead[pos].GetOrientation()))
                    {
                        me->CastSpell(target, SPELL_INFORM_DEAD_TRAINEE, true);
                    }
```

**Comment by Si1ker** [comment]
> Got your point, will do today but via C++ script not within db.

@Ragelezz ok.

**Comment by FrancescoBorzi** [comment]
what's the status of this?

**Comment by Ragelezz** [comment]
Not ongoing at the moment, visuals are there but its more of a hackfix since the souls needs to travel from point to point not npc>boss>destination.

**Comment by Si1ker** [comment]
It's better to open a new PR with the work suggested in the issue description, literally the solution is in there, just needs to be adjusted to the ac core. Please read it if you still want to work on it.
Closing as this is not even a hackfix.

**Pull request closed without merging**